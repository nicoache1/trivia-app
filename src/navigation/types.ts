import { ParamListBase } from '@react-navigation/native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import React from 'react'

import { Routes } from './routes'

export interface ParamList extends ParamListBase {
  [Routes.Welcome]: undefined
  [Routes.ConfigureGame]: undefined
  [Routes.Game]: undefined
  [Routes.Results]: undefined
  [Routes.AnimatedSplash]: undefined
}

export type SceneProps<T extends keyof ParamList> = React.FC<
  NativeStackScreenProps<ParamList, T>
>

export type NavigationProp<T extends keyof ParamList> = NativeStackScreenProps<
  ParamList,
  T
>['navigation']
