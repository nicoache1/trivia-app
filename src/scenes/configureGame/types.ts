import { Difficulties, QuestionAmounts } from '../../types/game'

export interface ConfigureGameForm {
  name: string
  amount: number
  difficulty: number
}
