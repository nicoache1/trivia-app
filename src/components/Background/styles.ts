import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  vector: {
    ...StyleSheet.absoluteFillObject,
    top: 70,
  },
})
