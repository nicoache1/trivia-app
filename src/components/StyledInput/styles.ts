import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  input: {
    backgroundColor: 'transparent',
  },
  labelContainer: {
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
})
