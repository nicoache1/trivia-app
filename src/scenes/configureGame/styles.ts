import { Dimensions, StyleSheet } from 'react-native'

const { width } = Dimensions.get('window')

export const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  formContainer: {
    flex: 4,
  },
  headerContainer: {
    alignItems: 'center',
    backgroundColor: 'transparent',
    justifyContent: 'center',
    width,
  },
  scene: {
    backgroundColor: 'transparent',
  },
  screenContainer: {
    flex: 1,
  },
  subtitle: {
    paddingVertical: 10,
  },
  textContainer: {
    flexDirection: 'row',
  },
  vector: {
    bottom: 0,
    left: 0,
    position: 'absolute',
    right: 0,
    top: 70,
  },
})
