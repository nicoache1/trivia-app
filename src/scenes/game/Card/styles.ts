import { Dimensions, StyleSheet } from 'react-native'

const { height } = Dimensions.get('window')

export const styles = StyleSheet.create({
  categoryContainer: {
    flex: 1,
  },
  categoryLabel: {
    borderRadius: 16,
    height: 40,
    justifyContent: 'center',
    minWidth: 120,
  },
  container: {
    borderRadius: 16,
    height: height * 0.5,
    overflow: 'hidden',
    padding: 20,
    position: 'absolute',
    width: '100%',
  },
  questionContainer: {
    flex: 2,
  },
  text: {
    textAlign: 'center',
  },
})
