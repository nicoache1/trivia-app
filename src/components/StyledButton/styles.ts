import { StyleSheet } from 'react-native'

import { ITheme } from '../../styles/types'
import { ButtonMode } from './ButtonMode'

export const textStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    color: theme.colors.WHITE,
  },
  [ButtonMode.LIGHT]: {
    color: theme.colors.WHITE,
  },
})

export const buttonStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    backgroundColor: theme.colors.PRIMARY,
  },
  [ButtonMode.LIGHT]: {
    backgroundColor: theme.colors.SECONDARY,
  },
})

export const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    borderRadius: 10,
    height: 56,
    justifyContent: 'center',
    paddingVertical: 10,
    width: '100%',
  },
  buttonText: {
    alignItems: 'center',
  },
})
