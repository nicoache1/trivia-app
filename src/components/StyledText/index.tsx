import { concat } from 'lodash'
import React from 'react'
import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { AppTypography, ColorPalette } from '../../styles/types'

type TextCustomProps = {
  typography?: AppTypography
  color?: ColorPalette
  children: React.ReactNode
} & TextProps

export const StyledText: React.FC<TextCustomProps> = props => {
  const { Theme: ActualTheme } = useTheme()
  const {
    color = ColorPalette.BLACK,
    children,
    typography = AppTypography.P1_REGULAR,
    style,
  } = props

  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.BLACK

  const themedTypography = ActualTheme.typography[typography]

  const contextStyle: StyleProp<TextStyle> = {
    color: themedColor,
    ...themedTypography,
  }

  const mergedStyle: StyleProp<TextStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <Text {...props} style={mergedStyle} accessible={true}>
      {children}
    </Text>
  )
}
