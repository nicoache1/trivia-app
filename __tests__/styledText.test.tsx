import { render } from '@testing-library/react-native'
import React from 'react'
import { StyledText } from 'src/components/StyledText'
import { AppTypography, ColorPalette } from 'src/styles/types'

describe('Test StyledText', () => {
  test('Expect StyledText to be rendered correctly with default', async () => {
    const children = 'Test'

    const renderedSnapshot = render(
      <StyledText>{children}</StyledText>,
    ).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })

  test('Expect StyledText to be rendered correctly with color & typography', async () => {
    const children = 'Test'

    const renderedSnapshot = render(
      <StyledText
        color={ColorPalette.BLUE_SURFACE}
        typography={AppTypography.P1_MEDIUM}>
        {children}
      </StyledText>,
    ).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })
})
