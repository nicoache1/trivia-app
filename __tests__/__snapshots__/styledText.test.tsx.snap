// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StyledText Expect StyledText to be rendered correctly with color & typography 1`] = `
<Text
  accessible={true}
  color="BLUE_SURFACE"
  style={
    Object {
      "color": "#98D0E8",
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "500",
    }
  }
  typography="P1_MEDIUM"
>
  Test
</Text>
`;

exports[`Test StyledText Expect StyledText to be rendered correctly with default 1`] = `
<Text
  accessible={true}
  style={
    Object {
      "color": "#000000",
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;
